source/blog/2021-11-26-dbal-3.2.0.md

```diff
@@ -0,0 +1,140 @@
+// new schema
+$column2 = new Column('contents', Type::getType('text'), ['default' => 'Hello, world!']);
+```
+
+If we compared them with the comparator, we'd get a diff:
+
+```php
+$comparator = new Comparator();
+$comparator->diffColumn($column1, $column2);
+// array(1) {
+//   [0] =>
+//   string(7) "default"
+// }
+```
+
+This might be valid for the platforms that support the `DEFAULT`  constraint on `TEXT` columns but isn't valid for those
+that don't support it (e.g. MySQL). Regardless of the diff, both definitions would produce the same DDL on MySQL:
+
+```sql
+contents LONGTEXT NOT NULL
+```
+
+An attempt to migrate the old schema to the new one would produce a false-positive diff but applying it wouldn't result
+in any schema changes.
+
+A false-negative diff was also possible. Consider these following example:
+
+```php
+// old schema
+$column1 = new Column('settings', Type::getType('json'));
+
+// new schema
+$column2 = new Column('settings', Type::getType('json'), ['length' => 16777215]);
+```
+
+Comparison of the above column definitions should have triggered a diff on MySQL and migrate the underlying column
+from `TEXT` to `MEDIUMTEXT` but it didn't, because the DBAL would ignore the length of the `TEXT` columns.
+
+Apart from that, the DBAL would compare only a subset of the definitions, so some column options as the character set
+and collation weren't taken into account during comparison at all.
+
+### The new approach
+
+Instead of comparing abstract definitions on a per-property basis, the new implementation compares the DDL that is generated from both definitions for the target database platform. If the definitions produce the same DDL, they are considered equal. According to the tests and the number of resolved issues, this approach should be more accurate and less error-prone.
+
+Implementing this approach was impossible without introducing a new API which rendered the existing API obsolete.
+
+Prior to DBAL `3.2.0`, the schema comparator could be only instantiated directly via the `new` keyword:
+
+```php
+$comparator = new Comparator();
+```
+
+Instantiated like this, the comparator doesn't have a notion of the target database platform and cannot perform the comparison properly. That is why, this way of instantiation is deprecated in favor of instantiating the comparator by the schema manager:
+
+```php
+$schemaManager = $connection->createSchemaManager();
+$comparator = $schemaManager->createComparator();
+```
+
+This way, the schema manager can instantiate a platform-specific comparator and provide it with the necessary context (e.g. the default collation used by the database).
+
+While the old API is still available, it is recommended to use the new API for more accurate comparison.
+
+## Support for `psr/cache` ([#4620](https://github.com/doctrine/dbal/pull/4620))
+
+Since the Doctrine Cache library is being sunset, the new DBAL relase introduced the ability to use a PSR-6 compatible implementation for result caching.
```

Comment:
```suggestion
Since the Doctrine Cache library is being sunset, the new DBAL release introduced the ability to use a PSR-6 compatible implementation for result caching.
```